### PR TITLE
Create GET /orders/invoice/:invoice_number to view order details and item list

### PR DESCRIPTION
**Pull request by CharlotteW923**
## Description

Allows admins to retrieve an order's details by invoice number.

## Changes

- Creates the GET /orders/invoice/:invoice_number endpoint
- Allows only admins to access this endpoint
- Retrieves the orders details when provided the invoice number

## Screenshots/demo

image
image
image

## Related issues

What issue(s) does this PR fix?
#47

### CODE REVIEW

File: api/services/orders/order_handler.go [hunk 1]
```diff
@@ -274,3 +275,24 @@ func isValidStatusUpdate(currentStatus string, newStatus string) error {
...
+// @Produce json
+// @Security Bearer
+// @Param invoice_number path string true "Invoice Number"
+// @Success 200 {object} types.OrderByInvoiceResponse
+// @Router /orders/invoice/{invoice_number} [get]
+func (h *OrderHandler) handleGetOrderByInvoice(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	invoiceNumber := vars["invoice_number"]
+
+	response, err := h.store.GetOrderByInvoice(invoiceNumber)
```
Comment (sebastian-nunez) [inline api/services/orders/order_handler.go:291]: nit; rename `response` to `order`

File: api/services/orders/order_handler.go [hunk 2]
```diff
@@ -274,3 +275,24 @@ func isValidStatusUpdate(currentStatus string, newStatus string) error {
...
+	if err != nil {
```
Comment (sebastian-nunez) [inline api/services/orders/order_handler.go:292]: I think you need to have two conditions:

1. If `err != nil`: this is an `InternalServerError` returning the `err`
2. If `order == nil`: this a `StatusNotFound` with the message you have

File: api/types/payloads.go [hunk 3]
```diff
@@ -85,3 +85,12 @@ type UpdateContactRequest struct {
 	Email     string `json:"email" validate:"required,email"`
 	Phone     string `json:"phone" validate:"required"`
 }
+
+type OrderByInvoiceResponse struct {
```
Comment (sebastian-nunez) [inline api/types/payloads.go:89]: You can remove this and just use the `Order` entity

File: api/services/orders/order_handler.go [hunk 4]
```diff
@@ -274,3 +275,24 @@ func isValidStatusUpdate(currentStatus string, newStatus string) error {
...
 	return fmt.Errorf("order status can not change from '%v' to '%v'", currentStatus, newStatus)
 }
+
+// @Summary Retrieve order details by invoice number
+// @Description Retrieves order details and item list by invoice number.
+// @Tags Orders
+// @Produce json
+// @Security Bearer
+// @Param invoice_number path string true "Invoice Number"
+// @Success 200 {object} types.OrderByInvoiceResponse
```
Comment (sebastian-nunez) [inline api/services/orders/order_handler.go:285]: This should be an `object` of `types.Order`

**Review by sebastian-nunez — COMMENTED** [review]
Good work!

File: api/services/orders/order_handler.go (hunk 1 above)
Comment (CharlotteW923) [reply api/services/orders/order_handler.go:291]: Fixed!

File: api/types/payloads.go (hunk 3 above)
Comment (CharlotteW923) [reply api/types/payloads.go:89]: Fixed!

File: api/services/orders/order_handler.go (hunk 4 above)
Comment (CharlotteW923) [reply api/services/orders/order_handler.go:285]: Fixed!

File: api/services/orders/order_handler.go (hunk 2 above)
Comment (CharlotteW923) [reply api/services/orders/order_handler.go:292]: Is this better?
image

Comment (sebastian-nunez) [reply api/services/orders/order_handler.go:292]: for the `err != nil` case you can just do `WriteError(w, ..., err)`

**Review by sebastian-nunez — APPROVED** [review]
LGTM

File: api/services/orders/order_handler.go (hunk 2 above)
Comment (CharlotteW923) [reply api/services/orders/order_handler.go]: image

**Pull request merged**